feat(config): Env overrides for all standard config values

forslund opened the pull request:
This adds env overrides for the missing config options (excluding "extras") and may possibly resolve #12.

To note: OVOS_BUS_USE_SSL will set the ssl-flag if the environment variable exists. The parsing of bools seemed a bit limited using the standard `.parse()` so instead of requiring it to be EXACTLY `true` this felt like a more safe way.

mikejgray commented on README.md:
````diff
@@ -102,7 +104,7 @@ To run the test suite:
 ```
 
-...except we don't have tests. Please feel free to contribute!
+...except we don't have very many tests. Please feel free to contribute!
````
+1 for accuracy

mikejgray reviewed (approved):
LGTM. I'm going to hold off on merging until I get an automated release mechanism in place, should be by EOW 🙂

mikejgray commented:
@forslund Looks like it's not compiling as-is? https://github.com/OscillateLabsLLC/ovos-rust-messagebus/actions/runs/11999520451/job/33578793820?pr=15

forslund commented:
I've fixed the clippy issues but got trouble with the tests running in parallell despite the "[serial]" attribute. (this worked last week!) so I also updated the CI-config to only use a single thread.

mikejgray commented:
> I've fixed the clippy issues but got trouble with the tests running in parallell despite the "[serial]" attribute. (this worked last week!) so I also updated the CI-config to only use a single thread.

Thanks for taking care of that. I got the automated release workflow in place but unfortunately it created a conflict. If you wouldn't mind resolving that I'll go ahead and merge this in and we'll release 1.1

forslund commented:
The tests seem to have been completely removed. I've added them back as a separate github action yaml
